build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -217,10 +229,22 @@ private List<CompilationDatabaseInformation> populateObjList(IProject project, I
 						outputLocation + "", inputStrings, sourceLocation, outputLocation); //$NON-NLS-1$
 
 				IBuildMacroProvider provider = ManagedBuildManager.getBuildMacroProvider();
-				String resolvedOptionFileContents = provider.resolveValueToMakefileFormat(cmdLInfo.getCommandLine(), "", //$NON-NLS-1$
-						" ", IBuildMacroProvider.CONTEXT_FILE, //$NON-NLS-1$
-						new FileContextData(sourceLocation, outputLocation, null, tool));
+				String compilerName = CompilationDatabaseGenerator.getCompilerName(tool);
+				String commandLine = cmdLInfo.getCommandLine();
+				commandLine = commandLine.replace(compilerName, "").trim(); //$NON-NLS-1$
+				String compilerPath = findCompilerInPath(tool, config);
+				String resolvedOptionFileContents;
+				if (compilerPath != null && !compilerPath.isEmpty()) {
+					resolvedOptionFileContents = provider.resolveValueToMakefileFormat(
+							findCompilerInPath(tool, config) + " " + commandLine, "", " ", //$NON-NLS-1$//$NON-NLS-2$//$NON-NLS-3$
```

Comment:
```suggestion
							compilerPath + " " + commandLine, "", " ", //$NON-NLS-1$//$NON-NLS-2$//$NON-NLS-3$
```

(you may need to reapply formatting too because of line length change